packages/react-i18n/docs/react_i18n_schema.md

```diff
@@ -0,0 +1,284 @@
+
+Prior to the creation of this document, the React I18n schema was not documented.
+With the creation of this document, the schema was specified and given the version identifier `1.0`.
+
+# Versioning
+
+The version identifier for the schema is made of two parts, `<major>` and `<minor>`, separated by a `.`: `<major>.<minor>`.
+
+the `<major>` version number will be incremented any time a change is introduced that would mean that existing parsers would break while parsing the new schema. Put another way, software that supports parsing version `x.y` of the schema will always be able to parse any version `x.z`, where `z >= y`.
+
+The `<minor>` version number will be incremented for any other changes to the schema and/or serialization formats. For example, changes that make the schema "more strict" in what is accepted as valid can be done as increments to the `<minor>` version number, since parsers of older versions are still able to parse the newer versions, albeit "more loosely" than what a modern parser would.
+
+# Basic Structure
+
+React I18n is made of nested dictionaries of key-value pairs (KVPs). Each of the keys are strings, and the values are either strings, or a nested dictionary.
+
+In TypeScript, the type is a recursive definition:
+
+```typescript
+export interface TranslationDictionary {
+  [key: string]: string | TranslationDictionary;
+}
+```
+
+When [serialized to JSON]](#serialization-to-json), each of the dictionaries become JSON `object` types, and the keys and leaf values are both JSON `string` types.
+
+**Example React I18n JSON file**
+
+```jsonc
+{
+  "hello": "My name is {name}",
+  "parent": {
+    "child_1": "I am the first child of `parent`!",
+    "child_2": {
+      "grandchild_1": "I am the first grandchild of `parent`!",
+      "grandchild_2": "I am the second grandchild of `parent`!",
+    }
+  }
+}
+```
+
+# Pluralization
+
+Pluralization is handled by providing the pluarlization keys required by the language as leaf KVPs, creating a "pluralization context".
+
+**Example English file with pluralization keys:**
+```jsonc
+{
+  "cars": {  // All the children of `cars` are in a "pluralization context"
+    "one": "I have {count} car",
+    "other": "I have {count} cars"
+  }
+}
+```
+
+Different languages require different sets of pluralization keys. For example, the Polish (`pl`) has 4 required pluralization keys: `one`, `few`, `many`, and `other`. The Polish version of the previous example would be:
+
+**Example Polish file with pluralization keys:**
+```jsonc
+{
+  "cars": {
+    "one": "Mam {count} samochód",
+    "few": "Mam {count} samochody",
+    "many": "Mam {count} samochodów",
+    "other": "Mam {count} samochodu",
+  }
+}
+```
+
+## Reserved pluralization keys
+
+In order to avoid collisions, the following keys are reserved for use as pluralization keys, and must not be used as keys of leaf KVPs outside of the pluralization context:
```

Comment:
```suggestion
In order to avoid name collisions, the following keys are reserved for use as pluralization keys and must not be used as keys of leaf KVPs outside of a pluralization context:
```

Reply:
Can I use reserved key names for internal nodes?

Reply:
> Can I use reserved key names for internal (non-leaf) nodes?

@sjahu, yes. AFAIK, there is no need to restrict usage in non-leaf nodes.

Reply:
Fixed.